fvm/flex/models/block.go

```diff
@@ -0,0 +1,92 @@
+package models
+
+import (
+	"encoding/binary"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+// FlexBlock represents a Flex block.
+// It captures block info such as height and state
+type FlexBlock struct {
+	// Height returns the height of this block
+	Height uint64
+
+	// UUID Index
+	UUIDIndex uint64
+
+	// holds the total amount of the native token deposited in Flex
+	TotalSupplyOfNativeToken uint64
+
+	// StateRoot returns the EVM root hash of the state after executing this block
+	StateRoot gethCommon.Hash
+
+	// EventRoot returns the EVM root hash of the events emitted during execution of this block
+	EventRoot gethCommon.Hash
+}
+
+const (
+	encodedUInt64Size = 8
+	encodedHashSize   = 32
+
+	// ( height + uuid + state root + event root)
+	encodedBlockSize = 8 + 8 + 8 + 32 + 32
+)
+
+func (b *FlexBlock) ToBytes() []byte {
```

Comment:
Just curious: where is the byte representation used? Is it stored in the Flow execution state?

Reply:
Yeah, I store it as part of the Flex account in the Flow execution state, we use it for tracking state roots etc. we only keep the latest executed block and not the history.